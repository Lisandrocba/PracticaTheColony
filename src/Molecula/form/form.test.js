import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import FormDate from "./form";

describe("Form component", () => {
  test("Snapshot form component", () => {
   const {asFragment} = render(<FormDate />)
   expect(asFragment()).toMatchSnapshot()
  });
  const setupName = () => {
    const utils = render(<FormDate />)
    const input = screen.getByLabelText('name-input')
    return {
      input,
      ...utils,
    }
  }
  test('input initialaicer', ()=>{
    const {input} = setupName()
    expect(input.value).toBe('')
  })
  test('updates form state on input change', ()=>{
    const {input} = setupName()
    fireEvent.change(input, {target: {value: 'lisandro'}})
    expect(input.value).toBe('lisandro')
  })
  const setupEmail = () => {
    const utils = render(<FormDate />)
    const input = screen.getByLabelText('email-input')
    return {
      input,
      ...utils,
    }
  }
  test('input email initialaicer', ()=>{
    const {input} = setupEmail()
    expect(input.value).toBe('')
  })
  test('updates form state on input email change', ()=>{
    const {input} = setupName()
    fireEvent.change(input, {target: {value: 'lisandro@hotmail.com'}})
    expect(input.value).toBe('lisandro@hotmail.com')
  })
});