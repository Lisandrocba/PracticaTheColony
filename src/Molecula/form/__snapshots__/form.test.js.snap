// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component Snapshot form component 1`] = `
<DocumentFragment>
  <form>
    <input
      aria-label="name-input"
      name="nombre"
      type="text"
      value=""
    />
    <input
      aria-label="email-input"
      name="email"
      type="email"
      value=""
    />
    <input
      type="submit"
    />
  </form>
</DocumentFragment>
`;
